academic_observatory_workflows/workflows/openalex_telescope.py

```diff
@@ -597,180 +597,239 @@ def aws_to_gcs_transfer(self, release: OpenAlexRelease, **kwargs):
                 bucket=self.aws_openalex_bucket, aws_key=self.aws_key, entity_name=entity.entity_name
             )
             if entity.manifest != current_manifest:
-                logging.error(f"aws_to_gcs_transfer: entity {entity.entity_name} manifests have changed")
+                logging.error(f"aws_to_gcs_transfer: OpenAlexEntity({entity.entity_name}) manifests have changed")
                 success = False
             if entity.merged_ids != current_merged_ids:
-                logging.error(f"aws_to_gcs_transfer: entity {entity.entity_name} merged_ids have changed")
+                logging.error(f"aws_to_gcs_transfer: OpenAlexEntity({entity.entity_name}) merged_ids have changed")
                 success = False
 
         if success:
             logging.info(f"aws_to_gcs_transfer: manifests and merged_ids the same")
         set_task_state(success, self.aws_to_gcs_transfer.__name__, release)
 
-    def transform(self, release: OpenAlexRelease, **kwargs):
+    def determine_branches(self, release: OpenAlexRelease, **kwargs):
+        """Determine which entity branches to follow"""
+
+        task_ids = []
+        for entity in release.entities:
+            task_ids.append(f"{entity.entity_name}.bq_snapshot")
+
+        return task_ids
+
+    def bq_snapshot(self, release: OpenAlexRelease, entity_name: str = None, **kwargs):
+        """Create a snapshot of each main table. The purpose of this table is to be able to rollback the table
+        if something goes wrong. The snapshot expires after self.snapshot_expiry_days."""
+
+        task_id = get_task_id(**kwargs)
+        if release.is_first_run:
+            raise AirflowSkipException(make_first_run_message(task_id))
```

Comment:
I've tried these skipexections before, but in my testing they seem to skip everything downstream as well. Did you look into this?

Reply:
Yep. Tasks have a default trigger rule of `TriggerRule.SUCCESS`, which means that direct downstream tasks will only run if their direct upstream task was successful. To make the downstream task run on success or skipped, then the `TriggerRule.NONE_FAILED` trigger rule needs to be used.

Reply:
Ah I see! The other tasks must have skipped (in my testing) because that's the default behaviour. I can see that you've altered the downstream task trigger rule accordingly.